### ISSUE TITLE
[MURRAY][PDX225] Failure to build AOSP 12 - kernel-pdx225 missing and no known rule to make it

### ISSUE THREAD

**Issue by norenh** [sonyxperiadev/bug_tracker#793]
**Platform:** Murray (Sony Xperia 10 IV)
**Device:** xqcc54
**Kernel version:** 5.4(?)
**Android version:** android-12.0.0_r28

**Previously working on**
This is a new issue AFAIK. Have not managed to build it.

**Description**
I tried building AOSP 12 for my Sony Xperia 10 IV, while running Ubuntu 22.04 LTS, according to the instructions at:
https://developer.sony.com/develop/open-devices/guides/aosp-build-instructions/build-aosp-android-12-0
I used branch: **android-12.0.0_r28** and picked **aosp_xqcc54-eng** in the _"source build/envsetup.sh && lunch"_-step.
However, when running "make -j2" I get the following error:

```
henning@nuc1:~/android$ make -j2
build/make/core/soong_config.mk:195: warning: BOARD_PLAT_PUBLIC_SEPOLICY_DIR has been deprecated. Use SYSTEM_EXT_PUBLIC_SEPOLICY_DIRS instead.
build/make/core/soong_config.mk:196: warning: BOARD_PLAT_PRIVATE_SEPOLICY_DIR has been deprecated. Use SYSTEM_EXT_PRIVATE_SEPOLICY_DIRS instead.
============================================
PLATFORM_VERSION_CODENAME=REL
PLATFORM_VERSION=12
TARGET_PRODUCT=aosp_xqcc54
TARGET_BUILD_VARIANT=eng
TARGET_BUILD_TYPE=release
TARGET_ARCH=arm64
TARGET_ARCH_VARIANT=armv8-2a
TARGET_CPU_VARIANT=generic
TARGET_2ND_ARCH=arm
TARGET_2ND_ARCH_VARIANT=armv8-2a
TARGET_2ND_CPU_VARIANT=generic
HOST_ARCH=x86_64
HOST_2ND_ARCH=x86
HOST_OS=linux
HOST_OS_EXTRA=Linux-5.15.0-53-generic-x86_64-Ubuntu-22.04-LTS
HOST_CROSS_OS=windows
HOST_CROSS_ARCH=x86
HOST_CROSS_2ND_ARCH=x86_64
HOST_BUILD_TYPE=release
BUILD_ID=SQ1A.220205.002
OUT_DIR=out
PRODUCT_SOONG_NAMESPACES=device/sony/common device/sony/murray vendor/qcom/opensource/core-utils vendor/qcom/opensource/display/sm8350 vendor/qcom/opensource/display-commonsys-intf/sm8
350
============================================
[ 91% 439/478] including system/sepolicy/Android.mk ...
system/sepolicy/Android.mk:57: warning: BOARD_PLAT_PUBLIC_SEPOLICY_DIR has been deprecated. Use SYSTEM_EXT_PUBLIC_SEPOLICY_DIRS instead.
system/sepolicy/Android.mk:62: warning: BOARD_PLAT_PRIVATE_SEPOLICY_DIR has been deprecated. Use SYSTEM_EXT_PRIVATE_SEPOLICY_DIRS instead.
FAILED: ninja: 'kernel/sony/msm-5.4/common-kernel/kernel-pdx225', needed by 'out/target/product/pdx225/kernel', missing and no known rule to make it
11:01:19 ninja failed with: exit status 1

#### failed to build some targets (05:21 (mm:ss)) ####
```

**Symptoms**
Failure to build AOSP according to official instructions

**How to reproduce**
Follow the instructions on https://developer.sony.com/develop/open-devices/guides/aosp-build-instructions/build-aosp-android-12-0

**Additional context**
Additional logs from my build attempt can be provided on request

Thank you in advance for looking into this!

**Comment by jerpelea** [comment]
Hi 
we are working on fixing the murray kernel and then it will be added to the Open Devices program 
If you want to help with the bring-up we are accepting contributions .
If you are not a kernel developer please wait for the blogpost before you start building.
Best regards
Alin

**Comment by jerpelea** [comment]
I know but it will be officially supported only after the blog post is published
The code is open and community kernel developers are welcome to help 

Best regards
Alin

**Comment by vicky5124** [comment]
I was able to get past the error with the following steps:

```bash
# the scripts are meant to run on bash, not zsh or other shells
# and nice to not blow up the PC while it's compiling
nice bash

cd /path/to/android

source build/envsetup.sh .
lunch # option 71 for aosp_xqcc54-userdebug

cd kernel/sony/msm-5.4/common-kernel/
./build-kernels-clang.sh -d pdx225
cd ../../../../

# `make` didn't work, use `m`
m -j$(nproc)
# or `m -j4` for systems with 16 GB of ram (or less?)

# All of this results in the files shown in the image below.
```

I have not tried to flash the images yet though, as the lack of a recovery image, as well as the substantial difference in kernel file size between the precompiled devices and the newly compiled one, worries me that the phone will enter a boot loop or brick itself.

![2022-12-24_19-01](https://user-images.githubusercontent.com/8344197/209446950-8df4d737-480d-4df4-86be-d914c9a3a01f.png)
